src/ini/hashmap/hashmap.h

```diff
@@ -0,0 +1,53 @@
+// Copyright 2020 Joshua J Baker. All rights reserved.
```

Comment:
Maybe we would be better using hashmap implemented in systemd for consistency? What do you think?
@sdunnagan @mwperina @engelmi @Darth-Mera

Reply:
As I have already written in https://github.com/engelmi/orch/pull/27#discussion_r1044358123:
using a (hash-)map on the top-level is too ambiguous for a contract (e.g. the ini file representation in our case). Just a simple struct is sufficient and clear - and solves this issue by removing the need for a map implementation.

Reply:
I have to say I'm not 100% sure on which side I'm on here. I like the idea of using map for configuration, but on the other hand you have a point too

Reply:
I'm favoring Michael's argument.  The hash map seems overkill for where we're at right now with the application.  The added abstraction of a hash map would be a good idea to avoid coupling versions of the application to the config.  But in this case, it seems like we want the strong contract with the user's of the application, and also failing early in the parsing stage if not all the mandatory config values are provided.  We could do cool things later with abstraction, for example automatic discovery of node topology.  But for now, it's expected that we would need a new version of the application when the config changes.

By the way, I think we should add a config value for application version.